src/demo.rs

```diff
@@ -731,22 +752,61 @@ where
                     tracing::info!("\t{}", opt);
                     values.push(opt);
                 }
-
                 self.votes.insert(options.voteidx, values);
+
+                // Replay shunted messages if we have them. This ensures that we don't print VoteCast events for Vote we haven't seen the
+                // VoteOptions event for yet.
+                if !self.shunted_messages.is_empty() {
+                    let mut indices: Vec<usize> = vec![];
+                    for (idx, (vote_idx, shunted_msg)) in
+                        <Vec<(u32, DemoMessage)> as Clone>::clone(&self.shunted_messages)
+                            .into_iter()
+                            .enumerate()
+                    {
+                        if vote_idx == options.voteidx || self.votes.contains_key(&vote_idx) {
+                            // We only care about VoteCast event messages in the shunted messages queue.
+                            if let DemoEvent::VoteCast(_, _) = shunted_msg.event {
+                                indices.push(idx);
+                            }
+                        }
+                    }
+                    // early escape if no shunted messages to handle
+                    if indices.is_empty() {
+                        break 'voteEvent;
+                    }
+                    // Sort and reverse as we are REMOVING elements from shunted_messages, which will IMPACT every idx after.
+                    // So we start with the highest idx and move backwards
+                    indices.sort_unstable();
+                    indices.reverse();
+                    tracing::info!("Recalled {} shunted messages.", indices.len());
```

Comment:
We shouldn't be printing this as info, I'm assuming this was intended for debugging to see if it actually worked? But we'd want to remove it or change it to debug level before merging.